CPS-XXXX/README.md

```diff
@@ -0,0 +1,152 @@
+---
+CPS: ?
+Title: Mutable shared state and atomicity are compatible
+Status: Open
+Category: Ledger
+Authors:
+    - Vladimir Kalnitsky <klntsky@gmail.com>
+Proposed Solutions: []
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/874
+Created: 2024-08-05
+License: CC-BY-4.0
+---
+
+## Abstract
+
+Full determinism of transactions, defined by the absence of shared mutable state, gives rise to valuable ledger properties, such as:
+
+- atomicity (a transaction is either fully accepted or fully rejected without a fee)
+- predictability of monetary changes and fees
+
+However, it comes at a cost:
+
+- The need to "re-introduce" non-determinism where it is required for the dApp logic (which is commonly done by using multiple transactions for a single user action) makes immediate execution of user actions impossible.
+- UTxO contention limits the number of concurrent users a dApp can have, because a single UTxO can be consumed by a single transaction only.
+
+Completely avoiding mutable shared state *may* be a suboptimal way to achieve the desirable properties, as the design space for alternatives has not been fully explored.
+
+## Problem
+
+It is impossible to build a dApp that has these three properties at the same time on a fully-deterministic ledger:
```

Comment:
I think the key to scaling solutions on Cardano without compromising its security and determinism lies in building layered solutions. Bridges and other scaling tech can help us batch transactions, reduce load on the main chain, and improve overall performance without giving up on the ledger determinism. In my view this is not a problem, but a feature of Cardano, and we need to find ways and adapt our solutions with that goal.